activerecord/lib/active_record/connection_adapters/sql_type_metadata.rb

```diff
@@ -29,6 +29,12 @@ def hash
         def attributes_for_hash
           [self.class, sql_type, type, limit, precision, scale]
         end
+
+      private
+
+        def intern(value)
+          value.is_?(String) ? -value : value
```

Comment:
Aren't these only called on String or nil? Would it be clearer with `value.nil? ? value : -value`?